### PR TITLE
Show server message on exception

### PR DESCRIPTION
**Pull request by seanses**
Depends on https://github.com/xetdata/xet-core/pull/391. Fix FUN-90.

Some tests on most used path:
```
(.env) di@di-mbp ~/xetdata/pyxet/python/pyxet % xet ls xet://localhost:3000:seanses-local/comp             
Repository is not accessible to free tier accounts. Upgrade your XetHub subscription to access private repositories: http://localhost:3000///subscription/upgrade.
(.env) di@di-mbp ~/xetdata/pyxet/python/pyxet % xet ls xet://localhost:3000:seanses-local/comp/main
Repository is not accessible to free tier accounts. Upgrade your XetHub subscription to access private repositories: http://localhost:3000///subscription/upgrade.
(.env) di@di-mbp ~/xetdata/pyxet/python/pyxet % xet ls xet://localhost:3000:seanses-local/comp/aomen
Repository is not accessible to free tier accounts. Upgrade your XetHub subscription to access private repositories: http://localhost:3000///subscription/upgrade.
(.env) di@di-mbp ~/xetdata/pyxet/python/pyxet % xet cp -r xet://localhost:3000:seanses-local/comp/main .
Repository is not accessible to free tier accounts. Upgrade your XetHub subscription to access private repositories: http://localhost:3000///subscription/upgrade.
(.env) di@di-mbp ~/xetdata/pyxet/python/pyxet % xet cp -r src xet://localhost:3000:seanses-local/comp/main        
Repository is not accessible to free tier accounts. Upgrade your XetHub subscription to access private repositories: http://localhost:3000///subscription/upgrade.
```

### CODE REVIEW

File: python/pyxet/pyxet/cli.py [hunk 1]
```diff
@@ -179,7 +179,8 @@ def ls(path: Annotated[str, typer.Argument(help="Source file or folder to list")
                 print(tabulate(listing, headers="keys"))
             return listing
         except Exception as e:
-            raise
+            print(f"{e}")
+            return
```
Comment (hoytak) [inline python/pyxet/pyxet/cli.py:183]: The problem with just return here is that the program as a whole will report success through the exit code.  We should probably use sys.exit(1)

File: python/pyxet/pyxet/file_system.py [hunk 2]
```diff
@@ -190,16 +190,17 @@ def branch_info(self, url):
...
+            if attr is None:
+                raise FileNotFoundError(
+                    f"Branch or repo not found, remote = {url_path.remote()}, branch = {url_path.branch}")
+            return {"name": url_path.name(),
                 "size": attr.size,
                 "type": attr.ftype}
-
+        except Exception as e:
+                print(f"{e}")
+                exit()
```
Comment (hoytak) [inline python/pyxet/pyxet/file_system.py:203]: We should use an error exit code.  Makes testing easier :-)

File: python/pyxet/pyxet/cli.py (hunk 1 above)
Comment (seanses) [reply python/pyxet/pyxet/cli.py:183]: Gotcha!

**Review by hoytak — APPROVED** [review]
LGTM

**Pull request merged**